src/plugins/intel_gpu/src/plugin/transformations/reduce_modification_for_false_keepdims.cpp

```diff
@@ -0,0 +1,115 @@
+                input = std::make_shared<ngraph::opset10::ReduceSum>(input, reduce_const, true);
+            else if (ngraph::is_type<ngraph::opset10::ReduceMean>(reduce))
+                input = std::make_shared<ngraph::opset10::ReduceMean>(input, reduce_const, true);
+            else if (ngraph::is_type<ngraph::opset10::ReduceMin>(reduce))
+                input = std::make_shared<ngraph::opset10::ReduceMin>(input, reduce_const, true);
+            else if (ngraph::is_type<ngraph::opset10::ReduceMax>(reduce))
+                input = std::make_shared<ngraph::opset10::ReduceMax>(input, reduce_const, true);
+            else if (ngraph::is_type<ngraph::opset10::ReduceProd>(reduce))
+                input = std::make_shared<ngraph::opset10::ReduceProd>(input, reduce_const, true);
+
+            input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name());
+            new_ops.push_back(input.get_node_shared_ptr());
+
+            // Reshape
+            auto reshape_shape = ngraph::Shape(input_shape.size(), 1);
+            // Expected that a feature axis is only un-reduced.
+            reshape_shape[0] = reduce_shape[0];
+            input = std::make_shared<ngraph::opset10::Reshape>(
+                input,
+                ngraph::opset10::Constant::create(ngraph::element::i64,
+                                                  ngraph::Shape{reshape_shape.size()},
+                                                  reshape_shape),
+                false);
+
+            input.get_node_shared_ptr()->set_friendly_name(reduce->get_friendly_name() + "_reshape_false_keepdims");
+            new_ops.push_back(input.get_node_shared_ptr());
+
+            ngraph::copy_runtime_info(reduce, new_ops);
+            reduce->output(0).replace(input);
+            return true;
+        }
+
+        return false;
+    };
+
+    auto m = std::make_shared<ngraph::pattern::Matcher>(reduce_pattern, "ModifyReduceForFalseKeepDims");
+    register_matcher(m, callback);
+}
+
+bool ModifyReduceForFalseKeepDims::is_unsupported_reordered_axes(std::vector<int64_t> reduce_axes,
+                                                                                size_t num_dim,
+                                                                                size_t num_spatial) {
+    bool is_remains_feature_axis = false;
+
+    // Case to reduce batch axis and spatial axes
+    if (reduce_axes.size() > 1 && count(reduce_axes.begin(), reduce_axes.end(), 0) != 0 &&
+        count(reduce_axes.begin(), reduce_axes.end(), 1) == 0) {
+        is_remains_feature_axis = true;
+        // Check if it reduces all spatial axes
+        for (size_t idx_spatial = (num_dim - num_spatial); idx_spatial < num_dim; idx_spatial++) {
+            if (count(reduce_axes.begin(), reduce_axes.end(), idx_spatial) == 0) {
+                is_remains_feature_axis = false;
+                break;
+            }
+        }
+    }
```

Comment:
can we make a common function to be used from here and `layout_optimizer`?

Reply:
I considered when I added exception, not only this case but also exception about reduce mode. Is,'t it weird?
If it looks fine, I will move this to primitives/reduce.hpp to be called from here and layout_optimizer.